API reference v1.23

feloy opened the pull request:
Changes to the API reference for the upcoming v1.23

sftim commented:
/milestone 1.23

jimangel commented:
@feloy thanks for getting a jump start on this! Is the plan to test it now and re-run it once SIG Release starts cutting release candidates (rc)?

feloy commented:
@jimangel my idea is to update this PR as soon as new alpha/beta/RC are released, as is it easier to incorporate API changes by small chunks.

reylejano commented:
/cc @jlbutler @mehabhalodiya @chrisnegus @ramrodo @nate-double-u

jlbutler commented:
/assign

sftim commented on content/en/docs/reference/kubernetes-api/workload-resources/horizontal-pod-autoscaler-v2.md:
```diff
@@ -1,11 +1,11 @@
 ---
 api_metadata:
-  apiVersion: "autoscaling/v2beta2"
```
Will this change mean that the docs for HorizontalPodAutoscaler `autoscaling/v2beta2` disappear? This might surprise readers given that lots of tools do use this API at `autoscaling/v2beta2`.

feloy replied:
Yes, this change will make it disappear. I will check if we can have both v2beta2 and v2 in documentation

feloy replied:
@sftim I've updated the doc to display v1, v2beta2 and v2 for HorizontalPodautoscaler:
https://deploy-preview-29668--kubernetes-io-vnext-staging.netlify.app/docs/reference/kubernetes-api/workload-resources/

The problem is that versions are not displayed in the titles for non beta versions, and thus it is difficult to distinguish between v1 and v2

What would we like to do?
1. have only v2 and v2beta2?
2. always display the versions in the title (this will display v1 for a lot of resources)

(2) will need to update the program building the documentation, where (1) only needs to modify the configuration in this repository.

sftim replied:
Could we countenance a manual tweak, post build, for just the HPA APIs to show v1 and v2 explicitly, possibly followed by a logic tweak to show the version any time there are multiple stable APIs?

If we go for the logic tweak, SIG Docs can look for someone to pick up that work (ie: it doesn't have to be you @feloy - you already help a lot here!)

sftim replied:
We should definitely aim to leave in docs for HPA v1; until the v1.23 release it's the API we officially recommend (v2beta2 is, of course, only beta). The day after that release folks would still expect to see docs for that v1 API available.

sftim replied:
What do you prefer @feloy ?

feloy replied:
> Could we countenance a manual tweak, post build, for just the HPA APIs to show v1 and v2 explicitly, possibly followed by a logic tweak to show the version any time there are multiple stable APIs?

> If we go for the logic tweak, SIG Docs can look for someone to pick up that work (ie: it doesn't have to be you @feloy - you already help a lot here!)

@sftim, Yes, it would be nice to have someone make some manual tweak after build with the necessary changes, so I can work on the logic one once I have a precise example of what to do.

jimangel commented:
@feloy checking in, anything needed before Tuesday's release that we need to draw attention to?

/cc @jlbutler

feloy commented:
> @feloy checking in, anything needed before Tuesday's release that we need to draw attention to?
> 
> /cc @jlbutler

No, this PR is up to date for v1.23 rc1.

jlbutler commented:
thanks @feloy

@sftim @kbhawkey @tengqm please review, thanks!

jlbutler commented:
/assign @sftim @kbhawkey @tengqm

reylejano commented:
Hi @sftim @kbhawkey @tengqm , 
When you get the chance, please review and give an `lgtm` or `approve` if appropriate

kbhawkey commented:
:eyes:

tengqm commented:
although RC1 is the near complete version of 1.23, I'd still prefer deferring this to the formal release.

kbhawkey commented:
Page previews:
https://deploy-preview-29668--kubernetes-io-vnext-staging.netlify.app/docs/reference/kubernetes-api/

Field introduced in 1.23:
https://deploy-preview-29668--kubernetes-io-vnext-staging.netlify.app/docs/reference/kubernetes-api/common-parameters/common-parameters/#fieldValidation

https://deploy-preview-29668--kubernetes-io-vnext-staging.netlify.app/docs/reference/kubernetes-api/workload-resources/pod-v1/#parameters-6
https://deploy-preview-29668--kubernetes-io-vnext-staging.netlify.app/docs/reference/kubernetes-api/workload-resources/cron-job-v1/

kbhawkey commented:
> although RC1 is the near complete version of 1.23, I'd still prefer deferring this to the formal release.

OK, that's fair. We can post another build as soon as the release is available.
/lgtm

jlbutler commented:
Ok that works. I'll check back once we have a release tag for `1.23.0` and we can go from there. Thanks all!

jlbutler commented:
The release tag v1.23.0 is published. Is someone available to refresh this PR with a release build?

@kbhawkey @feloy

jlbutler commented:
Hi @feloy :wave: 

1.23 is released and 1.23.0 tag is available. Can you rebuild/refresh this PR, and then rebase to main? Thanks!

feloy commented:
> Hi @feloy wave
> 
> 1.23 is released and 1.23.0 tag is available. Can you rebuild/refresh this PR, and then rebase to main? Thanks!

Hi @jlbutler I've updated to 1.23.0 tag, and rebased to main branch. Thanks

sftim commented:
/remove-language zh
/approve

sftim commented:
/remove-area blog

sftim commented:
I'd expect that https://deploy-preview-29668--kubernetes-io-main-staging.netlify.app/docs/reference/kubernetes-api/workload-resources/horizontal-pod-autoscaler-v2/ would work - and it doesn't.

Even so, I've no objection to these changes going live, and then further fixups.

feloy commented:
@sftim you can get a preview at https://deploy-preview-29668--kubernetes-io-vnext-staging.netlify.app/docs/reference/kubernetes-api/

kbhawkey commented:
Page preview:
https://deploy-preview-29668--kubernetes-io-vnext-staging.netlify.app/docs/reference/kubernetes-api/workload-resources/horizontal-pod-autoscaler-v2beta2/

kbhawkey commented:
Thanks @feloy .
/lgtm
